### ISSUE TITLE
Gradient of torch.acosh(-1.1) is a normal value instead of NaN

### ISSUE THREAD

**Issue by maybeLee** [pytorch/pytorch#94509]
### 🐛 Describe the bug

As stated by the documentation: https://pytorch.org/docs/stable/generated/torch.acosh.html, the input outside of [1, Inf) will be NaN. However, when calculating the gradient of torch.acosh(-1.1), its output is a normal value. Please run the following code to reproduce:
```
import torch

y=torch.tensor(-1.1, requires_grad=True)
res = torch.acosh(y)  # output: nan, gradient: normal value.
print(res)  # nan
grad = torch.autograd.grad(res, (y))
print(grad)  # expected output: nan, actual output: (tensor(2.1822),)
```

I tried a few values and find that this issue occurs when input value is smaller than -1. For your reference, in the TensorFlow's implementation, both the output and gradient of tf.math.acosh are NaN when the input is -1.1 or other values smaller than -1.

### Versions

PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (x86_64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.102)
CMake version: Could not collect
Libc version: N/A

Python version: 3.8.15 (default, Nov 24 2022, 09:04:07)  [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==1.13.0
[conda] numpy                     1.24.1                   pypi_0    pypi
[conda] torch                     1.13.0                   pypi_0    pypi

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

**Comment by albanD** [comment]
Hey!
This is expected I'm afraid: the gradient at points where the function is not defined can be arbitrary. We have quite a few functions that do that to improve speed.
You can find more details on this in the doc: https://pytorch.org/docs/master/notes/autograd.html#gradients-for-non-differentiable-functions

**Comment by maybeLee** [comment]
Hi @albanD,
Thanks for your explanation, I am closing this issue.
